classloader/src/main/java/io/smallrye/common/classloader/ClassDefiner.java

```diff
@@ -0,0 +1,64 @@
+package io.smallrye.common.classloader;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+
+public class ClassDefiner {
+    private static final Object unsafe;
+    private static final Method defineClass;
+
+    static {
+        unsafe = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+            public Object run() {
+                try {
+                    final Class<?> unsafeClass = Class.forName("sun.misc.Unsafe");
+                    final Field field = unsafeClass.getDeclaredField("theUnsafe");
+                    field.setAccessible(true);
+                    return field.get(null);
+                } catch (IllegalAccessException e) {
+                    throw new IllegalAccessError(e.getMessage());
+                } catch (NoSuchFieldException e) {
+                    throw new NoSuchFieldError(e.getMessage());
+                } catch (ClassNotFoundException e) {
+                    throw new NoClassDefFoundError(e.getMessage());
+                }
+            }
+        });
+
+        defineClass = AccessController.doPrivileged(new PrivilegedAction<Method>() {
```

Comment:
Why are we using reflection to call this method instead of calling it directly?

Reply:
The method was removed in JDK 11.

Reply:
But we build an MR JAR, and we build with `release=8`, so we don't have to care about that - but you *do* have to add the [`build-include-jdk-misc` file](https://github.com/smallrye/smallrye-parent#mr-jar-sun-misc) in order to make the build work.

Reply:
Let me check.

Reply:
+1 to create a MR JAR

Reply:
Ok done. I've added a custom fix here, because parent had a bug in the scope: https://github.com/smallrye/smallrye-parent/pull/156